src/go/k8s/pkg/resources/certmanager/schemaregistry_api.go

```diff
@@ -20,6 +20,13 @@ const (
 
 // SchemaRegistryAPINodeCert returns the namespaced name for the SchemaRegistry API certificate used by nodes
 func (r *PkiReconciler) SchemaRegistryAPINodeCert() types.NamespacedName {
+	tlsListener := r.pandaCluster.SchemaRegistryAPITLS()
+	if tlsListener != nil && tlsListener.TLS.NodeSecretRef != nil {
+		return types.NamespacedName{
+			Name:      tlsListener.TLS.NodeSecretRef.Name,
+			Namespace: r.pandaCluster.Namespace,
```

Comment:
we really need to refactor this because e.g. this line is relying on the fact that you properly call prepareAPI and the secret is copied to our namespace 🙈 this is not obvious for the reviewer and very error prone. It's not against this PR but we should make all these stuff a bit more obvious...

Reply:
maybe ideally the PKI should return list of certificates generated and then we should use that all places to inject to sts to avoid having these things hardcoded multiple places. WDYT? (I can open an issue)

Reply:
I would like to see change in passing `NamespacedName` on every resource. I'm 💯  for new issue that describe the problem

Reply:
> we really need to refactor this

💯 

> the PKI should return list of certificates generated and then we should use that all places to inject to sts to avoid having these things hardcoded multiple places

👍 

> I can open an issue

@alenkacz  please do

>I would like to see change in passing NamespacedName on every resource

@RafalKorepta could you elaborate?

Reply:
We are passing NamespacedName in the constructor of every resource to create dependency. IMHO this could be changed.

Reply:
@alenkacz I've created this issue (https://github.com/vectorizedio/redpanda/issues/3550) so that we can use it to put all the improvements we want to do to PKI and certificate management